DigitalLearningSolutions.Web/Controllers/LoginController.cs

```diff
@@ -176,6 +182,10 @@ bool rememberMe
                 IssuedUtc = DateTime.UtcNow
             };
             HttpContext.SignInAsync("Identity.Application", new ClaimsPrincipal(claimsIdentity), authProperties);
+            if (!string.IsNullOrEmpty(returnUrl))
+            {
+                return Redirect(returnUrl);
```

Comment:
I think this might be susceptible to open-redirect attacks. Would it be possible to implement one of the methods found here: https://docs.microsoft.com/en-us/aspnet/core/security/preventing-open-redirects?view=aspnetcore-3.1#protecting-against-open-redirect-attacks  

I'm thinking that the IsLocalUrl method, with some additional logging as described in those docs would be useful, but am unsure if the logging is something we really need.

Reply:
If we can get an alternative redirect in if it looks like an open-redirect attack, then it might be worth making a unit test for that situation.

Reply:
I've added a fix for this, but it breaks the tests due to them needing a real UrlHelper. I haven't yet been able to work out how to initialise a new one so the ticket isn't complete.